dm/dm/master/scheduler/scheduler.go

```diff
@@ -2131,39 +2066,56 @@ func (s *Scheduler) handleWorkerOnline(ev ha.WorkerEvent, toLock bool) error {
 	}
 
 	// 2. check whether is bound.
-	if w.Stage() == WorkerBound {
-		// also put identical relay config for this worker
-		if source := w.RelaySourceID(); source != "" {
-			_, err := ha.PutRelayConfig(s.etcdCli, source, w.BaseInfo().Name)
-			if err != nil {
-				return err
-			}
+	// also put identical relay config for this worker
+	relaySources := w.RelaySources()
+	bounds := make([]ha.SourceBound, 0, len(relaySources))
+	for source := range relaySources {
+		bounds = append(bounds, ha.SourceBound{
+			Source: source,
+			Worker: w.baseInfo.Name,
+		})
+	}
+	if len(bounds) > 0 {
+		_, err := ha.PutRelayConfig(s.etcdCli, bounds...)
+		if err != nil {
+			return err
```

Comment:
suddenly I think there should not return (but I don't know what we should do), because DM-master only have one chance to handle DM-worker online, and the worker is indeed online. We'll left a zombie DM-worker if we return for this etcd error

Reply:
Maybe we can do some retry here and try to revoke this lease to let dm-worker retrigger this job. I will leave a TODO here.